apps/genpkey.c

```diff
@@ -103,10 +103,36 @@ static void show_gen_pkeyopt(const char *algname, OSSL_LIB_CTX *libctx, const ch
     EVP_PKEY_CTX_free(ctx);
 }
 
+int write_to_file(BIO *in, const char *filename, int format, int private)
```

Comment:
This looks like a nice general function that can be useful to other OpenSSL apps.
I suggest renaming it to, e.g., `mem_bio_to_file()` (also in `apps/lib/apps.h` of course)
and moving it to `apps/lib/apps.c` in the vicinity of `bio_open_owner()`.

Reply:
Updated.